include/crpropa/massdistribution/ConstantDensity.h

```diff
@@ -0,0 +1,65 @@
+#ifndef CRPROPA_CONSTANTDENSITY_H
+#define CRPROPA_CONSTANTDENSITY_H
+
+#include "crpropa/Units.h"
+#include "crpropa/Vector3.h"
+#include "crpropa/massdistribution/Density.h"
+
+#include <math.h>
+#include <sstream>
+
+#include "kiss/logger.h"
+
+
+namespace crpropa {
+
+/*
+@class ConstantDensity 
+@brief Density module for Constant densitys in HI, HII and H2 component. 
+*/
+class ConstantDensity: public Density {
+
+private:
+	// default mode: all density types set to 0 and no activ component
+	double HIdensitynumber  = 0;	
+	double HIIdensitynumber = 0;
+	double H2densitynumber  = 0;
+	
+	bool isHI = false;
+	bool isHII = false;
+	bool isH2 = false;
+
+public:
+	ConstantDensity(double HI, double HII, double H2);
+	
+	double getDensity(const Vector3d &position) const;
+	
+	double getHIDensity(const Vector3d &position) const;
+	double getHIIDensity(const Vector3d &position) const;
+	double getH2Density(const Vector3d &position) const;
+	double getNucleonDensity(const Vector3d &position) const;
+	
+	bool getisforHI();
+	bool getisforHII();
+	bool getisforH2();
+	
+	void setHI(bool activate, double densitynumber);
```

Comment:
I must say that I don't understand why we need to activate a specific density and do all the connected steps and checks with it if a not-used density can be just 0 and be always activated? Or am I missing something?

Reply:
If you call the getDensity or the getNucleonDensity function, only active parts are summed up. 
If you call the get\<Type>Density function, the (constant)Densitynumber for this type is returned. 
So in the case that you want to ignore the ionised componend for the interaction you can deactivate this component. But if other modules (which we don't have jet) needs the charged part of the massdistribution, it is possible to call the getHIIDensity. So you don't need to have two densities for this case.